app/battle.py

```diff
@@ -0,0 +1,16 @@
+from app.knight import Knight
+
+
+class Battle:
+    def __init__(self, knight1: Knight, knight2: Knight) -> None:
+        self.knight1 = knight1
+        self.knight2 = knight2
+
+    def fight(self) -> dict:
+        self.knight1.take_damage(self.knight2.total_power)
+        self.knight2.take_damage(self.knight1.total_power)
```

Comment:
The `fight` method currently applies damage using the opponent's `total_power` directly. According to the task description, the damage should be calculated by subtracting the knight's `protection` from the opponent's `total_power`. Ensure that the `take_damage` method in the `Knight` class is correctly invoked to account for this logic.